### DISCUSSION TITLE
resolve contract avatar with “useEnsAvatar”

### DISCUSSION THREAD

**Discussion by peduarte** [wevm/wagmi#33] [Ideas]
i noticed that if an ens avatar points to a contract address, it won't resolve properly. (example https://twitter.com/peduarte/status/1478016174566871040?s=21)

apparently we can use:  `provider.getAvatar('<ens_name>')` to get the avatar url even when it's a contract. but i haven't tried 

also, i haven't checked the implementation yet of the hook yet, im about to go on holiday but wanted to create this before i forget

**Comment by o-az** [comment]
with users like `brantly.eth`, the avatar is a contract like this:

`eip155:1/erc721:0xb7F7F6C52F2e2fdb1963Eab30438024864c313F6/2430`

in this case `provider.getAvatar('brantly.eth')` will get you the image url. Same if you supply brantly's eth address.

Oddly, with `ryanouyang.eth` it doesn't work. Even though they both have the same type of contract in their ENS avatar and in both cases the hyperlink in app.ens.name points to OpenSea. Btw, this seems to be an `ethers.js` issue cause I tried with ethers's `provider`

**Reply by tmm** [reply]
Fixed if you upgrade to `ethers@5.5.4` https://github.com/ethers-io/ethers.js/pull/2271#issuecomment-1025334108

**Reply by clumsier** [reply]
Hey there! I'm still encountering an issue with those package versions:
```
"ethers": "^5.7.2",
"wagmi": "^0.10.5",
```

I get avatar for https://etherscan.io/enslookup-search?search=brantly.eth as `https://i.imgur.com/5njQAB8.png`

But for - https://etherscan.io/enslookup-search?search=j1mmy.eth I got `null`

Seems it still can't properly resolve `eip` stuff

**Reply by tmm** [reply]
Interesting! wagmi uses ethers here so this would be a question for them on why it doesn't work.
